SFINT-4341 Readme file updated (#27)
coveooss/sf-case-assist-cookbook: README.md

It requires a working deployment of a [Case Assist Configuration](https://docs.coveo.com/en/3328/service/manage-case-assist-configurations) to provide [Document Suggestions](https://docs.coveo.com/en/3328/service/manage-case-assist-configurations#configuring-the-document-suggestion-functionality) as well as [Case Classifications](https://docs.coveo.com/en/3328/service/manage-case-assist-configurations#configuring-the-case-classification-functionality).

## Table of Contents

- [What Is Included in This Project](#what-is-included-within-this-project): A description of the contents of this repository.

- [How to Deploy](#how-to-deploy): This is the recommended installation option. Use this option if you want to see the app and the code in action.

- [How to Add New Fields for Classification](#how-to-add-new-fields-for-classification): A guide to specify additional fields for classification.

## What Is Included in This Project

This project contains two flows: a Recommended Flow and a Demo flow. Both are lightning flows already setup with multiple steps to create a support case.

### Recommended Flow:

The Recommended Flow contains the following screens:

> 

Author: Yeah sure!

1. A first screen where the user can enter a Subject and a Description for their case and can see the strength of this Description using the Description Strength Indicator.
2. A second screen where the user can find predictions to help classify their case. This screen will predict values for the Case Priority, Case Type, and Case Reason fields as these are Standard Salesforce Case fields.
3. A third screen where documents will be suggested to the user based on the text entered in the case Subject and Description. The user can read more about each document suggestion with the help of the Quickview and can also leave their feedback on each document.
4. A confirmation screen that the case was created.

### Demo Flow:

The Demo Flow contains the following screens:

> 1. A first screen where the user can enter a Subject and a Description for their case and can see the strength of this Description, using the Description Strength Indicator, as well as the predictions to help classify their case. The user can see the predictions for the Case Priority, Case Type, and Case Reason fields as they type in the Subject and Description inputs.

Reviewer: See "How to Add New Fields for Classification" section to learn how to modify these.

2. A second screen where documents will be suggested to the user based on the text entered in the case Subject and Description. The user can read more about each document suggestion with the help of the Quickview and can also leave their feedback on each document.
3. A confirmation screen that the case was created.

PS: A Login screen is implemented in both flows. It's just a template that you can use to provide a custom login screen to access the case assist flow.

## How to Deploy

> ### 1. Set up your environment

Reviewer 2: ```suggestion
### 1. Set up Your Environment
```

> These steps assume you already have the [Coveo for Salesforce package installed](https://docs.coveo.com/en/1158/coveo-for-salesforce/getting-started-with-coveo-for-salesforce) and configured (linked to a Coveo organization).

### 2. Install the Quantic Library

[Install the Coveo Quantic Library as a Salesforce unlocked package](https://docs.coveo.com/en/quantic/latest/usage/#install-quantic).

### 3a. Deploy the Project Using the Org Development Model

The Org Development Model allows you to connect directly to a non-source-tracked org (sandbox, Developer Edition (DE) org, Trailhead Playground, or even a production org) to retrieve and deploy code directly. This model is similar to the type of development you have done in the past using tools such as Force.com IDE or MavensMate.

To start developing with this model in Visual Studio Code, see [Org Development Model with VS Code](https://forcedotcom.github.io/salesforcedx-vscode/articles/user-guide/org-development-model). For details about the model, see the [Org Development Model](https://trailhead.salesforce.com/content/learn/modules/org-development-model) Trailhead module.

> When working with non-source-tracked orgs, use the commands `SFDX: Deploy Source to Org` (VS Code) or `sfdx force:source:deploy` (Salesforce CLI) and `SFDX: Retrieve Source from Org` (VS Code) or `sfdx force:source:retrieve` (Salesforce CLI). The `Push` and `Pull` commands work only on orgs with source tracking (scratch orgs).

Reviewer: Is there a way we could write in a step by step way? Like: 
1. Open your CLI tool
2. Navigate to X
3. Enter the command SFDX: Deploy Source to Org
4. etc.

Author: Yes it'll be easier to get started, I'll work on this

### 3b. Push the Code in a Scratch Org

When working with source-tracked orgs, use the commands `SFDX: Push Source to Org` (VS Code) or `sfdx force:source:push` (Salesforce CLI) and `SFDX: Pull Source from Org` (VS Code) or `sfdx force:source:pull` (Salesforce CLI). Do not use the `Retrieve` and `Deploy` commands with scratch orgs.

### 3c. Install the App Using an Unlocked Package

TBD

### 4. Enable the Case Flow in Your Community

> 1. In your Salesforce community, drag the Lightning Flow component in a Community page and select the `Case_Assist_Recommended_Flow` or the `Case_Assist_Demo_Flow` shipped with this repository.

Reviewer: How will a user "select" one or the other? It is in a modal and they will literally select an option from a dropdown?

Author: Exactly yeah, after adding the flow component to a community page, the following modal will appear allowing the user to select the flow they wanna use:
<img width="398" alt="Flow_Modal" src="https://user-images.githubusercontent.com/86681870/154678853-a9c6fe87-1d13-404a-9114-f7bc24758920.png">

> 2. After selecting the name of the flow, two inputs labeled `caseAssistId` and `engineId` appear. Fill the `caseAssistId` input with your [Case Assist Id](https://docs.coveo.com/en/3328/#retrieving-a-case-assist-id) retrieved from your Case Assist Configuration and fill the `engineId` input with an appropriate name for the engine that the Quantic components will register to.

Reviewer: Is there any doc we can refer to regarding the Quantic engineID?

Author: Didn’t find any 🧐

Reviewer 3: This [Usage](https://docs.coveo.com/en/quantic/latest/usage/#javascript) section mentions it but doesn't describe it.
This reference doc for the [QuanticCaseAssistInterface](https://docs.coveo.com/en/quantic/latest/reference/case-assist-components/case-assist-case-assist-interface/) has a little description for the argument.

Author: Thanks! We can use that👌

To start developing with this model in Visual Studio Code, see [Org Development Model with VS Code](https://forcedotcom.github.io/salesforcedx-vscode/articles/user-guide/org-development-model). For details about the model, see the [Org Development Model](https://trailhead.salesforce.com/content/learn/modules/org-development-model) Trailhead module.

When working with non-source-tracked orgs, use the commands `SFDX: Deploy Source to Org` (VS Code) or `sfdx force:source:deploy` (Salesforce CLI) and `SFDX: Retrieve Source from Org` (VS Code) or `sfdx force:source:retrieve` (Salesforce CLI). The `Push` and `Pull` commands work only on orgs with source tracking (scratch orgs).

### 3b. Pushing the code in a Scratch Org

When working with source-tracked orgs, use the commands `SFDX: Push Source to Org` (VS Code) or `sfdx force:source:push` (Salesforce CLI) and `SFDX: Pull Source from Org` (VS Code) or `sfdx force:source:pull` (Salesforce CLI). Do not use the `Retrieve` and `Deploy` commands with scratch orgs.

### 3c. Installing the app using an Unlocked Package

> TBD

Reviewer 2: Noticing that this has been here for a while, I wonder whether we wouldn't want to just remove this section?

Reviewer: Agreed let's remove it for now, because this isn't an unlocked package, correct?

> ### 4. How to enable the Case Flow in your community

Reviewer 2: ```suggestion
### 4. Enable the Case Flow in Your Community
```

> 1. In your Salesforce community, drag the Lightning Flow component onto a Community page and select the Case_Assist_Recommended_Flow or the Case_Assist_Demo_Flow shipped with this repository.

Reviewer 2: ```suggestion
1. In your Salesforce community, drag the Lightning Flow component in a Community page and select the `Case_Assist_Recommended_Flow` or the `Case_Assist_Demo_Flow` shipped with this repository.
```

> 2. After selecting the name of the flow two inputs labeled caseAssistId and engineId appear. Fill the caseAssistId input with your [Case Assist Id](https://docs.coveo.com/en/3328/service/manage-case-assist-configurations#retrieving-a-case-assist-id "Case Assist Id") retrieved from your Case assist Configuration and fill the enginId input with an appropriate name for the engine that the Quantic components will register to.

Reviewer 2: ```suggestion
2. After selecting the name of the flow, two inputs labeled `caseAssistId` and `engineId` appear. Fill the `caseAssistId` input with your [Case Assist Id](https://docs.coveo.com/en/3328/#retrieving-a-case-assist-id) retrieved from your Case Assist Configuration and fill the `engineId` input with an appropriate name for the engine that the Quantic components will register to.
```

> 3. In the published version of your community, fill in the subject and description fields in the first screen.
> 4. Proceed to the next screens to see the proposed predictions for classifying your case and get documents suggestions that will potentially resolve your case before it is created.

Reviewer 2: ```suggestion
3. In the published version of your community, users can now fill in the subject and description fields in the first screen. If they proceed to the next screens to see the proposed predictions for classifying their case and get documents suggestions that will potentially resolve their case before it is created.
```
(I think that's clearer, since these last two steps are more about the outcome of `Enabling the Case Flow in Your Community` than they are steps to do so.)

### Dev, Build and Test

1. Run `npm i`

> 1. Run `npm run build`

Reviewer 2: I would also capitalize the title below:
`Developing locally` -> `Developing Locally`

### Testing

TBD

> ## How to Add New fields for Classification

Reviewer 2: ```suggestion
## How to Add New Fields for Classification
```

This cookbook suggests classifications for the standard case fields Case Priority, Case Type, and Case Reason.

> To specify additional fields for classification, access the [Provide Details Screen](src/main/default/lwc/provideDetailsScreen) and copy the sample code that are commented out in the `provideDetailsScreen.html` and `provideDetailsScreen.js` files as many times as necessary. Remember to replace the placeholders `<SALESFORCE_API_FIELD_NAME>` and `<COVEO_FIELD_NAME>` with the Salesforce API name and the Coveo field name of the new field to predict, respectively.

Reviewer: Add a sentence: 
You can find the Coveo Field Name in the [Fields ](https://docs.coveo.com/en/2036/index-content/about-fields) section of the Coveo Admin Console.

### Testing

TBD

> ## How to add new fields for classification

Reviewer 2: ```suggestion
## How to Add New fields for Classification
```

This cookbook suggests classifications for the standard case fields Case Priority, Case Type, and Case Reason.

> To specify additional fields for classification, access the [Provide Details Screen](src/main/default/lwc/provideDetailsScreen "Provide Details Screen") and copy the sample code that are commented out in the `provideDetailsScreen.html` and `provideDetailsScreen.js` files as many times as necessary. Remember to replace the placeholders `<SALESFORCE_API_FIELD_NAME>` and `<COVEO_FIELD_NAME>` with the Salesforce API name and the Coveo field name of the new field to predict, respectively.

Reviewer 2: ```suggestion
To specify additional fields for classification, access the [Provide Details Screen](src/main/default/lwc/provideDetailsScreen) and copy the sample code that are commented out in the `provideDetailsScreen.html` and `provideDetailsScreen.js` files as many times as necessary. Remember to replace the placeholders `<SALESFORCE_API_FIELD_NAME>` and `<COVEO_FIELD_NAME>` with the Salesforce API name and the Coveo field name of the new field to predict, respectively.
```

Reviewer 2: (unless there's a new link syntax I'm not aware of 🙂)

> Make sure that all newly added fields are specified in your configuration. For more information, see https://docs.coveo.com/en/3328/service/manage-case-assist-configurations#configuring-the-case-classification-functionality.

Reviewer 2: ```suggestion
Make sure that all newly added fields are specified in your configuration (see [Configuring the Case Classification Functionality](https://docs.coveo.com/en/3328/#configuring-the-case-classification-functionality)).
```
